src/api/app/controllers/webui/users_controller.rb

```diff
@@ -79,24 +85,24 @@ def update
       end
     end
 
-    if @configuration.accounts_editable?(@displayed_user)
-      @displayed_user.assign_attributes(params[:user].slice(:realname, :email).permit!)
-      @displayed_user.toggle(:in_beta) if params[:user][:in_beta]
-    end
+    assign_common_user_attributes if @configuration.accounts_editable?(@displayed_user)
+    assign_admin_attributes if User.admin_session?
 
-    if User.admin_session?
-      @displayed_user.assign_attributes(params[:user].slice(:state, :ignore_auth_services).permit!)
-      @displayed_user.update_globalroles(Role.global.where(id: params[:user][:role_ids])) unless params[:user][:role_ids].nil?
-    end
-
-    begin
-      @displayed_user.save!
-      flash[:success] = "User data for user '#{@displayed_user.login}' successfully updated."
-    rescue ActiveRecord::RecordInvalid => e
-      flash[:error] = "Couldn't update user: #{e.message}."
+    respond_to do |format|
+      if @displayed_user.save
```

Comment:
This block could be simplified like this:

```ruby
      format.any(:html, :js) do
        if @displayed_user.save
          flash[:success] = "User data for user '#{@displayed_user.login}' successfully updated."
        else
          flash[:error] = "Couldn't update user: #{@displayed_user.errors.full_messages.to_sentence}."
        end

        redirect_back(fallback_location: user_path(@displayed_user)) if request.format.symbol == :html
      end
```

The `flash[:error]` is the same for both cases, this is the only difference.

Reply:
I think your proposal is not exactly what we need, as for `js` format we need `flash.now`. I've refactored the code a bit, please have a look.

Reply:
Oh right, this is true!